tests/cpp_unit_tests/test_distribution.cpp

```diff
@@ -0,0 +1,63 @@
+#include "openmc/distribution.h"
+#include "openmc/random_lcg.h"
+#include <catch2/catch_test_macros.hpp>
+#include <pugixml.hpp>
+
+TEST_CASE("Test alias method sampling of a discrete distribution")
+{
+  int n_samples = 1e6;
+  double x[5] = {-1.6, 1.1, 20.3, 4.7, 0.9};
+  double p[5] = {0.2, 0.1, 0.5, 0.05, 0.15};
```

Comment:
You might want to add another test / modify this test to check the actual values rather than the mean. That is, confirm that if you sample 1,000,000 values, about 200,000 of them correspond to x[0].